src/Extensions/Logging/Internal/OmexLogsEventSource.cs

```diff
@@ -0,0 +1,277 @@
+			string category,
+			string tagId,
+			string tagName,
+			int threadId,
+			string message) =>
+			WriteEvent((int)EventSourcesEventIds.LogVerboseEventId, applicationName, serviceName, agentName, buildVersion, processName, partitionId, replicaId, correlationId, transactionId, level, category, tagId, tagName, threadId, message);
+
+
+		[Event((int)EventSourcesEventIds.LogSpamEventId, Level = EventLevel.LogAlways, Message = "{13}", Version = 6)]
+		private void LogSpamServiceMessage(
+			string applicationName,
+			string serviceName,
+			string agentName,
+			string buildVersion,
+			string processName,
+			Guid partitionId,
+			long replicaId,
+			string correlationId,
+			string transactionId,
+			string level,
+			string category,
+			string tagId,
+			string tagName,
+			int threadId,
+			string message) =>
+			WriteEvent((int)EventSourcesEventIds.LogSpamEventId, applicationName, serviceName, agentName, buildVersion, processName, partitionId, replicaId, correlationId, transactionId, level, category, tagId, tagName, threadId, message);
+
+
+		/// <summary>
+		/// Get the Tag id as a string
+		/// </summary>
+		/// <param name="tagId">tag id</param>
+		/// <returns>the tag as string</returns>
+		/// <remarks>
+		/// In terms of the conversion from integer tag value to equivalent string reprsentation, the following scheme is used:
+		/// 1. If the integer tag &lt;= 0x0000FFFF, treat the tag as special tag called numeric only tag.
+		/// Hence the string representation is direct conversion i.e. tag id 6700 == 6700
+		/// 2. Else, if it's an alphanumeric tag, there are 2 different schemes to pack those. viz. 4 letter and 5 letter representations.
+		/// 2.1 four letter tags are converted by transforming each byte into it's equivalent ASCII. e.g. 0x61626364 => abcd
+		/// 2.2 five letter tags are converted by transforming lower 30 bits of the integer value into the symbol space a-z,0-9.
+		/// The conversion is done by treating each group of 6 bits as an index into the symbol space a,b,c,d, ... z, 0, 1, 2, ....9
+		/// eg. 0x000101D0 = 00 000000 000000 010000 000111 010000 2 = aaqhq
+		/// (from http://office/15/howto/reliability/Wiki/Assert%20and%20ULS%20Tagging.aspx)
+		/// </remarks>
+		private static string TagIdAsString(int tagId) // Convert should be done more efficient
+		{
+			if (tagId <= 0xFFFF)
+			{
+				// Use straight numeric values
+				return tagId.ToString("x4", CultureInfo.InvariantCulture);
+			}
+			else if (tagId <= 0x3FFFFFFF)
+			{
+				// Use the lower 30 bits, grouped into 6 bits each, index into
+				// valuespace 'a'-'z','0'-'9' (reverse order)
+				char[] chars = new char[5];
+				for (int i = 4; i >= 0; i--)
+				{
+					int charVal = tagId & 0x3F;
+					tagId = tagId >> 6;
+					if (charVal > 25)
+					{
+						if (charVal > 35)
+						{
+							chars[i] = '?';
+						}
+						else
+						{
+							chars[i] = (char)(charVal + 22);
+						}
+					}
+					else
+					{
+						chars[i] = (char)(charVal + 97);
```

Comment:
97 => 'a'
apply to the whole file

Reply:
I would like to maintain as many similarities with old code as possible, until we will delete one version, sorry